config/src/main/java/com/typesafe/config/ConfigException.java

```diff
@@ -442,6 +442,22 @@ public BadBean(String message) {
         }
     }
 
+    /**
+     * Exception indicating that a key was messed up, for example you may have
+     * asked for an Enum and the key text value does not exist on the requested Enum.
+     */
+    public static class BadKey extends BugOrBroken {
```

Comment:
We don't usually use BugOrBroken for what could be a mistake in the config file itself ... I'd probably go with extending ConfigException itself as with BadValue

Reply:
Sounds good to me, I will change it based on that ;)
Thanks